src/js/autocomplete.js

```diff
@@ -1,14 +1,247 @@
+import accessibleAutocomplete from 'accessible-autocomplete';
+
+/**
+ * @param {string} suggestion - Text which is going to be suggested to the user
+ * @param {string} query - Text which was typed into the autocomplete by the user
+ * @returns {[string, boolean][]} An array of arrays which contain two items, the first is the character in the suggestion, the second is a boolean which indicates whether the character should be highlighted.
+ */
+function highlightSuggestion(suggestion, query) {
+	const result = suggestion.split('');
+
+	const matchIndex = suggestion.toLocaleLowerCase().indexOf(query.toLocaleLowerCase());
+	return result.map(function(character, index) {
+		let shoudHighlight = true;
+		const hasMatched = matchIndex > -1;
+		const characterIsWithinMatch = index >= matchIndex && index <= matchIndex + query.length - 1;
+		if (hasMatched && characterIsWithinMatch) {
+			shoudHighlight = false;
+		}
+		return [character, shoudHighlight];
+	});
+}
+
+/**
+ * Create the DOM tree which corresponds to
+ * <div class="o-autocomplete__menu-loading-container">
+ * 	<div class="o-autocomplete__menu-loading"></div>
+ * </div>
+ * @returns {HTMLDivElement} The loading container as a HTMLDivElement
+ */
+function createLoadingContainer() {
+	const loadingContainer = document.createElement('div');
+	loadingContainer.classList.add('o-autocomplete__menu-loading-container');
+	const loading = document.createElement('div');
+	loading.classList.add('o-autocomplete__menu-loading');
+	loadingContainer.appendChild(loading);
+	return loadingContainer;
+}
```

Comment:
Non-blocking suggestion. I was curious if we could improve this function to remove the need for the DOM tree comment and think we can 😄 Could maybe do a similar thing for `createClearButton` too

```suggestion
/**
 * Create DOM for the loading container.
 * @returns {HTMLDivElement} The loading container.
 */
function createLoadingContainer() {
	return document.createRange().createContextualFragment(`
		<div class="o-autocomplete__menu-loading-container">
			<div class="o-autocomplete__menu-loading"></div>
		</div>
	`).querySelector(':first-child'); // IE does not support firstElementChild
}
```

Reply:
Nice!